src/tools/generate/generate_chapters.js

```diff
@@ -72,10 +125,67 @@ const write_template = async (language, year, chapter, metadata, body, toc) => {
   });
 
   await fs.outputFile(path, fomatted_html, 'utf8');
-
   await size_of(path);
 };
 
+const write_sitemap_start = async (language, year) => {
+  
+  await fs.outputFile(sitemap_path, '<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">\n', 'utf8');
+
+  await write_sitemap_for_file('templates/' + language + '/' + year + '/index.html', 'https://almanac.httparchive.org/en/2019/index.html');
+  await write_sitemap_for_file('templates/' + language + '/' + year + '/table_of_contents.html', 'https://almanac.httparchive.org/en/2019/table_of_contents.html');
+  await write_sitemap_for_file('templates/' + language + '/' + year + '/methodology.html', 'https://almanac.httparchive.org/en/2019/methodology.html');
+  await write_sitemap_for_file('templates/' + language + '/' + year + '/contributors.html', 'https://almanac.httparchive.org/en/2019/contributors.html');
+
+};
+
+const write_sitemap_for_file = async (file, url) => {
+  
+  var lastmod;
+
+  let static_file = fs.readFileSync(file).toString();
+  var rx = /"datePublished": "([0-9\-\+\:T]*)"/;
```

Comment:
For this, `datePublished` is being generated by the `published` frontmatter attribute on the md file. If we just use the data from the md file, then we can forego the regex. Would that be ok?

Reply:
Oh wait, nevermind. This is only for the static files.

Reply:
Yeah there must be a better way to do this too, but that's the best way I could come up with for now.

Reply:
Alternative would be to do all the .md files like this so only one way of doing it?